lib/rabl/builder.rb

```diff
@@ -147,7 +147,7 @@ def attribute(name, options = {})
       # node(:foo) { "bar" }
       # node(:foo, :if => lambda { |m| m.foo.present? }) { "bar" }
       def node(name, options = {}, &block)
-        return unless resolve_condition(options)
+        return unless resolve_condition(options.merge({ :name => name }))
```

Comment:
@Sharangovich can we do the same thing for `#child, `#glue` and `#extends`?

Reply:
I wouldn't do so because the `name` attribute for 
- `#child` is defined implicitly
- `#glue` and `#extends` is not defined at all
My recommendation would be to state the `name` attribute in the options hash for the mentioned methods.
Just like:
```ruby
child(@user, :name => :author)
or 
glue(@user, :name => :author)
```
I think that if you use it properly, then `resolve_condition` will be invoked with a name attribute from the block (`#attribute` or `#attributes` methods):
```ruby
child(@user) { attribute :full_name }
```
Please correct me if I'm missing something.